util/patch/suite_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package patch
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/klog"
+	"k8s.io/klog/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+	"sigs.k8s.io/cluster-api/test/helpers"
+	// +kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+func init() {
+	klog.InitFlags(nil)
+	log.SetLogger(klogr.New())
+}
+
+const (
+	timeout = time.Second * 10
+)
+
+var (
+	testEnv *helpers.TestEnvironment
+	ctx     = context.Background()
+)
+
+func TestPatch(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func(done Done) {
+	By("bootstrapping test environment")
+	var err error
+	testEnv, err = helpers.NewTestEnvironment()
+	Expect(err).NotTo(HaveOccurred())
+
+	By("starting the manager")
+	go func() {
+		Expect(testEnv.StartManager()).To(Succeed())
```

Comment:
I think you should add `defer GinkgoRecover()` when calling expectations from a go routine
```suggestion
		defer GinkgoRecover()
		Expect(testEnv.StartManager()).To(Succeed())
```

Reply:
This is copied from the other suite files, should we change them in a different PR all together?

Reply:
Separate PR works if this is a common problem